src/transformers/models/musicgen/modeling_musicgen.py

```diff
@@ -572,6 +572,24 @@ def forward(
                 output_attentions=output_attentions,
             )
 
+        # Ignore copy
+        if (
+            attention_mask is not None
+            and (attention_mask.mean(dim=[1, 2, 3]) <= torch.finfo(attention_mask.dtype).min).any()
+        ):
+            logger.warning_once(
+                '`torch.nn.functional.scaled_dot_product_attention` does not support having an empty attention mask. Falling back to the manual attention implementation. This warning can be removed using the argument `attn_implementation="eager"` when loading the model.'
+                "Note that this probably happens because `guidance_scale>1` or because you used `get_unconditional_inputs`. See https://github.com/huggingface/transformers/issues/31189 for more information."
+            )
+            return super().forward(
+                hidden_states,
+                key_value_states=key_value_states,
+                past_key_value=past_key_value,
+                attention_mask=attention_mask,
+                layer_head_mask=layer_head_mask,
+                output_attentions=output_attentions,
+            )
```

Comment:
Rather than fallback, I would just raise an exception. Otherwise this expensive check and forward pass can easily go unnoticed

Reply:
Well, `sdpa` and `guidance_scale>1` are used by default, so we'd raise the error almost every time. Also, even if the model uses eager mode for the cross-attention layers (in which the bug happens), it'll still benefit from the speed-up of the self-attention layers. 
 
Should we find a better way of testing the attention mask ? For example, we could raise a warning that it will happens [here](https://github.com/huggingface/transformers/blob/2965b204593df9d5652313386ec280ffbfd1753b/src/transformers/models/musicgen/modeling_musicgen.py#L2457-L2460) and [here](https://github.com/huggingface/transformers/blob/2965b204593df9d5652313386ec280ffbfd1753b/src/transformers/models/musicgen/modeling_musicgen.py#L2948) and switch the cross-attention SDPA layers to eager layers ?

Reply:
OK, I see. I'm a bit concerned about this causing unexpected behaviour, in particular defaulting to eager like this as it's a bit magic. As there's other layers which can still use SDPA, this seems like a pragmatic solution. 

Let's leave as-is. If more users raise issues, then we'll have to re-think.